src/test/java/swinglib/JComboBoxBuilderTest.java

```diff
@@ -0,0 +1,125 @@
+package swinglib;
+
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.swing.JComboBox;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.Is;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import games.strategy.triplea.settings.ClientSetting;
+
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class JComboBoxBuilderTest {
+  @Mock
+  private ItemEvent mockItemEvent;
+
+  @Before
+  public void setUp() {}
+
+  @Test(expected = IllegalArgumentException.class)
+  public void buildInvalidEmptyOptionSet() {
+    JComboBoxBuilder.builder()
+        .menuOptions()
+        .build();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void builderNoMenuOptionSpecified() {
+    JComboBoxBuilder.builder()
+        .build();
+  }
+
+  @Test
+  public void basicBuilderWithMenuOptions() {
+    final JComboBox box = JComboBoxBuilder.builder()
```

Comment:
Static analysis: "JComboBox is a raw type. References to generic type JComboBox<E> should be parameterized"